src/grpc.rs

```diff
@@ -75,3 +86,158 @@ impl ClientPolicies for Server {
+            }
+        }),
+    }
+}
+
+fn to_filters(policies: &PolicySet) -> Result<Vec<proto::Filter>, tonic::Status> {
+    let filters = policies
+        .policies
+        .values()
+        .flat_map(|spec| &spec.filters)
+        .map(convert_filter)
+        .collect::<Result<_, _>>()?;
+    Ok(filters)
+}
+
+fn convert_filter(filter: &Filter) -> Result<proto::Filter, tonic::Status> {
+    match filter {
+        Filter::Timeout(t) => {
+            let t = t.clone().try_into().map_err(|e| {
+                tonic::Status::internal(format!("Failed to convert timeout duration: {}", e))
+            })?;
+
+            Ok(proto::Filter {
+                filter: Some(proto::filter::Filter::Timeout(t)),
+            })
+        }
+    }
+}
+
+fn convert_match(
+    HttpRouteMatch {
+        headers,
+        path,
+        query_params,
+        method,
+    }: &HttpRouteMatch,
+) -> http_route::HttpRouteMatch {
+    let headers = headers
+        .into_iter()
+        .map(|hm| match hm {
+            HeaderMatch::Exact(name, value) => http_route::HeaderMatch {
+                name: name.to_string(),
+                value: Some(http_route::header_match::Value::Exact(
+                    value.as_bytes().to_vec(),
+                )),
+            },
+            HeaderMatch::Regex(name, re) => http_route::HeaderMatch {
+                name: name.to_string(),
+                value: Some(http_route::header_match::Value::Regex(re.to_string())),
+            },
+        })
+        .collect();
+
+    let path = path.as_ref().map(|path| http_route::PathMatch {
+        kind: Some(match path {
+            PathMatch::Exact(path) => http_route::path_match::Kind::Exact(path.clone()),
+            PathMatch::Prefix(prefix) => http_route::path_match::Kind::Prefix(prefix.clone()),
+            PathMatch::Regex(regex) => http_route::path_match::Kind::Regex(regex.to_string()),
+        }),
+    });
+
+    let query_params = query_params
+        .into_iter()
+        .map(|qpm| match qpm {
+            QueryParamMatch::Exact(name, value) => http_route::QueryParamMatch {
+                name: name.clone(),
+                value: Some(http_route::query_param_match::Value::Exact(value.clone())),
+            },
+            QueryParamMatch::Regex(name, re) => http_route::QueryParamMatch {
+                name: name.clone(),
+                value: Some(http_route::query_param_match::Value::Regex(re.to_string())),
+            },
+        })
+        .collect();
+
+    http_route::HttpRouteMatch {
+        headers,
+        path,
+        query_params,
+        method: method.clone().map(Into::into),
+    }
+}
+
+fn convert_rule(
+    client_policies: &PolicySet,
+    InboundHttpRouteRule { matches, filters }: &InboundHttpRouteRule,
+) -> Result<proto::http_route::Rule, tonic::Status> {
+    Ok(proto::http_route::Rule {
+        matches: matches.iter().map(convert_match).collect(),
+        filters: to_filters(client_policies)?,
```

Comment:
this needs to convert only the filters defined by ClientPolicies with a ClientPolicyBinding that selects the client pod's labels. see my comment on `to_filters` for details.